snappy/click.go

```diff
@@ -190,87 +203,83 @@ func verifyStructStringsAgainstWhitelist(s interface{}, whitelist *regexp.Regexp
 	return nil
 }
 
-func verifyServiceYaml(service ServiceYaml) error {
-	return verifyStructStringsAgainstWhitelist(service, servicesBinariesStringsWhitelist)
-}
-
-func generateSnapServicesFile(service ServiceYaml, baseDir string, aaProfile string, m *packageYaml) (string, error) {
-	if err := verifyServiceYaml(service); err != nil {
+func generateSnapServicesFile(app *AppYaml, baseDir string, aaProfile string, m *snapYaml) (string, error) {
+	if err := verifyAppYaml(app); err != nil {
 		return "", err
 	}
 
 	udevPartName := m.qualifiedName(originFromBasedir(baseDir))
 
-	desc := service.Description
+	desc := app.Description
 	if desc == "" {
-		desc = fmt.Sprintf("service %s for package %s", service.Name, m.Name)
+		desc = fmt.Sprintf("service %s for package %s", app.Name, m.Name)
 	}
 
 	socketFileName := ""
-	if service.Socket {
-		socketFileName = filepath.Base(generateSocketFileName(m, service))
+	if app.Socket {
+		socketFileName = filepath.Base(generateSocketFileName(m, app))
 	}
 
 	return systemd.New(dirs.GlobalRootDir, nil).GenServiceFile(
 		&systemd.ServiceDescription{
 			AppName:        m.Name,
-			ServiceName:    service.Name,
+			ServiceName:    app.Name,
 			Version:        m.Version,
 			Description:    desc,
 			AppPath:        baseDir,
-			Start:          service.Start,
-			Stop:           service.Stop,
-			PostStop:       service.PostStop,
-			StopTimeout:    time.Duration(service.StopTimeout),
+			Start:          app.Command,
+			Stop:           app.Stop,
+			PostStop:       app.PostStop,
+			StopTimeout:    time.Duration(app.StopTimeout),
 			AaProfile:      aaProfile,
 			IsFramework:    m.Type == snap.TypeFramework,
-			IsNetworked:    service.Ports != nil && len(service.Ports.External) > 0,
-			BusName:        service.BusName,
-			Forking:        service.Forking,
+			IsNetworked:    app.Ports != nil && len(app.Ports.External) > 0,
+			BusName:        app.BusName,
+			Forking:        app.Forking,
```

Comment:
Why not just implement `app.Daemon` which should hold valid `systemd` values (whitelisted `simple`, `forking`, ...) and use that as a replacement string?